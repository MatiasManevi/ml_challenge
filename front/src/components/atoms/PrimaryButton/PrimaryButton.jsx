import React from 'react';

import './PrimaryButton.scss';

const PrimaryButton = ({ text, clickHandler }) => {
	return (
		<button onClick={clickHandler} className="primary-button">
			{text}
		</button>
	);
};

export default PrimaryButton;
