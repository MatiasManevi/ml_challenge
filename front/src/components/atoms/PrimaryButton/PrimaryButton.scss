@import 'assets/scss/variables';

.primary-button {
	color: $color-white;
	font-size: $font-size-16;
	background-color: $color-secondary;
	width: 100%;
	height: $space-42;
	cursor: pointer;
	text-decoration: none;
	padding: $space-8;
	border-radius: $thumbnail-radius;
	border: none;
}
