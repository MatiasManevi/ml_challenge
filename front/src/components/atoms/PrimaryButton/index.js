export { default } from 'components/atoms/PrimaryButton/PrimaryButton';
