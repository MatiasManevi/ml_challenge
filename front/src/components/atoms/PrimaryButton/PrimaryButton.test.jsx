import React from 'react';
import { shallow } from 'enzyme';

import PrimaryButton from './PrimaryButton';

describe('PrimaryButton', () => {
	const clickHandler = jest.fn();

	it('should match snapshot', () => {
		const wrapper = shallow(
			<PrimaryButton text="Compra" clickHandler={clickHandler} />
		);
		expect(wrapper).toMatchSnapshot();
	});

	it('should call clickHandler when clicking', () => {
		const wrapper = shallow(<PrimaryButton text="Ok" clickHandler={clickHandler} />);
		wrapper.simulate('click');
		expect(clickHandler).toHaveBeenCalledTimes(1);
	});
});
