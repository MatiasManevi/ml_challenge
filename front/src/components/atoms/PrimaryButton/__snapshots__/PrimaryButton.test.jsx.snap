// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton should match snapshot 1`] = `
<button
  className="primary-button"
  onClick={[MockFunction]}
>
  Compra
</button>
`;
